rust/bin-shared/src/tun_device_manager.rs

```diff
@@ -30,6 +37,7 @@ mod tests {
         // Run these tests in series since they would fight over the tunnel interface
         // if they ran concurrently
         create_tun();
+        no_packet_loops().await;
```

Comment:
In #6220, I made this one an "Integration test" (`tests/`) because those always run in sequence. I'd prefer that over combining them into one test here.

Thoughts? It does mean a bit more linking as those are all individual binaries.

Reply:
We'd get more useful output if multiple tests fail if they're integration tests, right?
I may give it a shot in another PR, since it's just `bin-shared` it may link fast enough. If it was the GUI Client I'd be reluctant since it takes so long to link anything there

Reply:
@ReactorScram What do you think of this? It would be nice to get rid of this hack eventually.

Reply:
> We'd get more useful output if multiple tests fail if they're integration tests, right?

Yeah that would be the idea.